src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObject = {};
+  const sortDate = [];
+  const [, , , separatorFrom] = fromFormat;
+  const [, , , separatorTo] = toFormat;
+  const separateDate = date.split(separatorFrom);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY') {
+      if (separateDate[i] < 30) {
+        dateObject[fromFormat[i] + 'YY'] = '20' + separateDate[i];
+      } else {
+        dateObject[fromFormat[i] + 'YY'] = '19' + separateDate[i];
+      }
+    } else if (fromFormat[i] === 'YYYY' && toFormat.includes('YY')) {
+      dateObject['YY'] = separateDate[i].slice(2);
```

Comment:
The assignment to `dateObject['YY']` should be inside the loop iterating over `fromFormat`, and it should use the current index instead of a fixed 'YY' string. This would handle cases where 'YYYY' is not the first element in the `fromFormat` array.